### PR TITLE
fix: lending repayment don't try and estimate Tx fees on approval needed

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Does what it says on the box - no query, no problem

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/7083

> What protocols, transaction types or contract interactions might be affected by this PR?

Low, see testing

## Testing

- Token repayments with approval needed are now happy
- Token repayments without approval needed are still happy
- Native asset repayments are still happy

### Engineering

- ^

### Operations

- ^

## Screenshots (if applicable)

Develop (right) vs. this diff

- Approval required

image

- Approval granted

image

### CODE REVIEW

File: src/pages/Lending/Pool/components/Repay/RepayInput.tsx [hunk 1]
```diff
@@ -398,9 +405,12 @@ export const RepayInput = ({
       // no explicit from address required for repayments
       fromAddress: '',
       action: 'repayLoan',
+      enableEstimateFees: isApprovalRequired === false,
     },
   )
 
+  console.log({ isEstimatedFeesDataLoading })
+
```
Comment (gomesalexandre) [inline src/pages/Lending/Pool/components/Repay/RepayInput.tsx:413]: uh oh

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
✅ Native asset repay
✅ Already-approved asset repay
❌ Not-yet approved asset repay - getting strange behaviour here, see below:

On initial load I can approve the token, but the network fee never loads:

Screenshot 2024-06-11 at 3 37 06 PM

If I approve a token, and then go to a different, non-approved token, the approve button is disabled, and the network fee does not load:

Screenshot 2024-06-11 at 3 34 25 PM

https://jam.dev/c/afc17501-2e4f-4545-9c5e-b11e8f6d5755

**Comment by gomesalexandre** [comment]
> ✅ Native asset repay ✅ Already-approved asset repay ❌ Not-yet approved asset repay - getting strange behaviour here, see below:
> 
> On initial load I can approve the token, but the network fee never loads:
> 
> Screenshot 2024-06-11 at 3 37 06 PM
> If I approve a token, and then go to a different, non-approved token, the approve button is disabled, and the network fee does not load:
> 
> Screenshot 2024-06-11 at 3 34 25 PM
> https://jam.dev/c/afc17501-2e4f-4545-9c5e-b11e8f6d5755

ty for the review @0xApotheosis!

There are two components to this:

1. Network fee being shown as a skeleton, as we *cannot* estimate fees with an approval required. I believe we should bring in the same "network fees *or* approval fees pattern as in RFOX here" though that's probably a different concern IMO, going to bring this one in as a direct follow-up on top of this PR not to mix concerns. 
2. Infinite loading state while selecting another token after approving one. Indeed, there's definitely a case of wanting to clear the approvalTx to handle the edge case of having a stale approval Tx.

Brought a few more fixes in, which effectively contribute to fixing 2. (and more issues spotted whilst in the house):

- 01fa0d8d00 - ensures `repaymentAmountCryptoPrecision` is in the repayment asset's precision, *without* more precision as originally coming over from the wire (THOR). This means approvals would fail to be properly detected, as we're effectively approving a smaller amount (repayment asset precision on-chain) compared to the input amount (too much precision)
- 7203359e5c - uses `selectPortfolioCryptoPrecisionBalanceByFilter` to select the crypto precision balance precision, effectively fixing the wrong repayment asset balance, previously in base unit
- 1e3c4b1ebe - type-safety ensuring we have a quote when doing balance checks 

https://jam.dev/c/2e907761-f031-4102-bd30-4230032b0c9b

**Review by 0xApotheosis — APPROVED** [review]
This works as it claims to 🤝 

We still have the issue of the network fee skeleton on approval (i.e. not showing the approval fee estimate to the user), but that is not expected for this PR (though as you noted, we should absolutely do in a follow up).

**Pull request merged**